packages/clis/parser/gdeflate/gdeflate.cc

```diff
@@ -0,0 +1,49 @@
+#include <napi.h>
+#include "libdeflate.h"
+
+template<>
+struct std::default_delete<libdeflate_gdeflate_decompressor> {
+    void operator()(libdeflate_gdeflate_decompressor* p) const {
+        libdeflate_free_gdeflate_decompressor(p);
+    }
+};
+
+Napi::Value GDeflate(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+
+  if (info.Length() != 2) {
+    Napi::TypeError::New(env, "Wrong number of arguments")
+        .ThrowAsJavaScriptException();
+    return env.Null();
+  }
+  if (!info[0].IsArrayBuffer() || !info[1].IsArrayBuffer()) {
+    Napi::TypeError::New(env, "Arguments must be array buffers").ThrowAsJavaScriptException();
+    return Napi::Value::From(env, -1);
```

Comment:
If we're throwing a JS exception above, then the return value probably doesn't matter.

```suggestion
    return env.Null();
```